ui/components/app/snaps/update-snap-permission-list/update-snap-permission-list.js

```diff
@@ -37,75 +46,103 @@ export default function UpdateSnapPermissionList({
   );
 
   const snapsMetadata = useSelector(getSnapsMetadata);
-  const snapsNameGetter = getSnapName(snapsMetadata);
 
-  const approvedPermissionsToShow = {
+  const approvedCombinedPermissions = {
     ...approvedPermissions,
     connection_permission: approvedConnections ?? {},
   };
 
-  const revokedPermissionsToShow = {
+  const revokedCombinedPermissions = {
     ...revokedPermissions,
     connection_permission: revokedConnections ?? {},
   };
 
-  const newPermissionsToShow = {
+  const newCombinedPermissions = {
     ...newPermissions,
     connection_permission: newConnections ?? {},
   };
 
+  const [showAll, setShowAll] = useState(false);
+
+  const newWeightedPermissions = getWeightedPermissions({
+    t,
+    permissions: newCombinedPermissions,
+    subjectName: snapName,
+    getSubjectName: getSnapName(snapsMetadata),
+  });
+
+  const revokedWeightedPermissions = getWeightedPermissions({
+    t,
+    permissions: revokedCombinedPermissions,
+    subjectName: snapName,
+    getSubjectName: getSnapName(snapsMetadata),
+  });
+
+  const approvedWeightedPermissions = getWeightedPermissions({
+    t,
+    permissions: approvedCombinedPermissions,
+    subjectName: snapName,
+    getSubjectName: getSnapName(snapsMetadata),
+  });
+
+  const filteredApprovedWeightedPermissions = getFilteredSnapPermissions(
+    approvedWeightedPermissions,
+    PermissionWeightThreshold.snapUpdateApprovedPermissions,
+  );
+
+  if (Object.keys(approvedWeightedPermissions).length < 1 && !showAll) {
+    setShowAll(true);
```

Comment:
Lets use the condition to seed `useState´ instead of setting it here.

Reply:
Updated.